File: backend/api/schemas/price.py
from pydantic import BaseModel, Field


class PriceCreate(BaseModel):
    service_id: int
    title: str
    price: str
    position: int = Field(default=0, ge=0)
    is_hidden: bool = False


class PriceUpdate(BaseModel):
    title: str
    price: str
    position: int = Field(default=0, ge=0)
    is_hidden: bool = False


class PriceResponse(BaseModel):
    id: int
    service_id: int
    title: str
    price: str
    position: int
    is_hidden: bool

File: frontend/src/types/price.ts
export interface PriceCreate {
  service_id: number
  title: string
  price: string
  position?: number
  is_hidden?: boolean
}

export interface PriceUpdate {
  title: string
  price: string
  position?: number
  is_hidden?: boolean
}

export interface PriceResponse {
  id: number
  service_id: number
  title: string
  price: string
  position: number
  is_hidden: boolean
}

export interface PublicPriceResponse {
  id: number
  title: string
  price: string
  position: number
}